model/task_annotations/task_annotations.go

```diff
@@ -10,14 +10,10 @@ type TaskAnnotation struct {
 	Id            string `bson:"_id" json:"id"`
 	TaskId        string `bson:"task_id" json:"task_id"`
 	TaskExecution int    `bson:"task_execution" json:"task_execution"`
-	// comment about the failure
-	Note string `bson:"note,omitempty" json:"note,omitempty"`
-	// links to tickets definitely related.
-	Issues []IssueLink `bson:"issues,omitempty" json:"issues,omitempty"`
-	// links to tickets possibly related
-	SuspectedIssues []IssueLink `bson:"suspected_issues,omitempty" json:"suspected_issues,omitempty"`
-	// annotation attribution
-	Source AnnotationSource `bson:"source" json:"source"`
+	// the task annotation that is api sourced and will be overwritten when updated
+	APIAnnotation Annotation `bson:"api_annotation,omitempty" json:"api_annotation,omitempty"`
```

Comment:
omitempty only works for pointers(see here https://github.com/evergreen-ci/evergreen/pull/4093) , so you'll want to use *Annotation (also in the API)

Reply:
you checkmarked this here annotation is still not a pointer ?